runtime/lua/vim/treesitter/query.lua

```diff
@@ -748,45 +904,53 @@ end
 
 --- Iterates the matches of self on a given range.
 ---
---- Iterate over all matches within a {node}. The arguments are the same as
---- for |Query:iter_captures()| but the iterated values are different:
---- an (1-based) index of the pattern in the query, a table mapping
---- capture indices to nodes, and metadata from any directives processing the match.
---- If the query has more than one pattern, the capture table might be sparse
---- and e.g. `pairs()` method should be used over `ipairs`.
---- Here is an example iterating over all captures in every match:
+--- Iterate over all matches within a {node}. The arguments are the same as for
+--- |Query:iter_captures()| but the iterated values are different: an (1-based)
+--- index of the pattern in the query, a table mapping capture indices to a list
+--- of nodes, and metadata from any directives processing the match.
+---
+--- You MUST set the "correct" option for this function to behave correctly!
+--- In older versions of Nvim this function has incorrect behavior: the correct
+--- behavior must be opted into for backward compatibility.
```

Comment:
```suggestion
--- WARNING: Set `correct=true` to ensure all matching nodes in a match are
--- returned, otherwise only the last node in a match is returned, breaking captures
--- involving quantifiers such as `(comment)+ @comment`. The default option
--- `correct=false`is only provided for backward compatibility and will be removed
--- after Nvim 0.10.
```